### PR TITLE
Initial support for Istio VirtualService and DestinationRule resource types

### PR DESCRIPTION
**Pull request by olensmar**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [X] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/redux/thunks/previewCluster.ts [hunk 1]
```diff
@@ -181,7 +181,11 @@ async function loadCustomResourceObjects(kc: KubeConfig, customResourceDefinitio
       // retrieve objects using latest version name
       // @ts-ignore
       k8sApi
-        .listClusterCustomObject(r.content.spec.group, findDefaultVersion(r) || 'v1', r.content.spec.names.plural)
+        .listClusterCustomObject(
+          r.content.spec.group,
```
Comment (devcatalin) [inline src/redux/thunks/previewCluster.ts:185]: should we use conditional chaining on `r.content` as well? or check if this is not undefined

File: src/kindhandlers/istio/DestinationRule.handler.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import {
+  ISTIO_DEFAULT_RESOURCE_VERSION,
+  ISTIO_RESOURCE_GROUP,
+  ISTIO_SUBSECTION_NAME,
+} from '@src/kindhandlers/istio/constants';
+
+const DestinationRuleHandler = createCustomObjectKindHandler(
+  'DestinationRule',
+  ISTIO_SUBSECTION_NAME,
+  'DestinationRules',
```
[the remark this answers is not in the document]
Comment (olensmar) [reply src/kindhandlers/istio/DestinationRule.handler.ts:11]: let's skip the spaces - i've fixed those that were inconsistent

File: src/kindhandlers/common/customObjectKindHandler.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    clusterApiVersion: `${kindGroup}/${kindVersion}`,
+    sourceEditorOptions: pathToSchemaResource
+      ? {
+          editorSchema: loadCustomSchema(pathToSchemaResource, kind),
+        }
+      : undefined,
+    getResourceFromCluster(kubeconfig: k8s.KubeConfig, resource: K8sResource): Promise<any> {
+      const ix = resource.version.lastIndexOf('/');
+      const version = ix > 0 ? resource.version.substring(ix + 1) : kindVersion;
+      const group = ix > 0 ? resource.version.substring(0, ix) : kindGroup;
```
[the remark this answers is not in the document]
Comment (olensmar) [reply src/kindhandlers/common/customObjectKindHandler.ts]: done - thanks - perhaps we should rename `K8sResource.version` to `apiVersion` - to make clear it comes from that property !?

**Review by devcatalin — APPROVED** [review]
Great work!

**Pull request merged**